### PR TITLE
define hook maturity model

### LINKED ISSUE #195: What's the process for creating new hooks?
In an offline discussion with @brynrhodes, @isaacvetter, and @brettmarquard, we discussed the need for a better process for suggesting, reviewing, testing and publishing new hooks.

A few goals:
- Hooks should be able to be created, reviewed, published outside of the more rigid publishing calendar of the actual, core specification.
- Hooks should be tested before being published at FHIR Connectathons. The [FHIR Maturity Model](http://wiki.hl7.org/index.php?title=FHIR_Maturity_Model) would be a good model to follow.
- Hooks should be reviewed with an overall, holistic perspective to avoid redundancy and ensure new hooks follow best practices.
- Requestors should document the proposed hook on a new wiki page (following the [template](http://cds-hooks.org/hooks/template/)), link to the new wiki page from [Proposed Hooks](https://github.com/cds-hooks/docs/wiki/Proposed-Hooks) and create a new issue.

### PR DESCRIPTION
**Pull request by isaacvetter**
Fixes #195

### CODE REVIEW

File: docs/hooks/index.md [hunk 1]
```diff
@@ -147,6 +147,20 @@ Hook creators SHOULD include examples of the context.
 
 If the context contains FHIR data, hook creators SHOULD include examples across multiple versions of FHIR if differences across FHIR versions are possible.
 
+## Hook Maturity Model
+The intent of the CDS Hooks Maturity Model is to attain broad community engagement and consensus, before a hook is labeled as mature, that the hook is necessary, implementable, and worthwhile to the CDS services and CDS clients that would reasonably be expected to use it. Implementer feedback should drive the maturity of new hooks. Diverse participation in open developer forums and events, such as HL7 FHIR Connectathons, is necessary to achieve significant implementer feedback. The below criteria will be evaluated with these goals in mind.
+
+The Hook maturity levels use the term CDS client to generically refer to the clinical workflow system in which a CDS services returned cards are displayed.
```
Comment (kpshek) [inline docs/hooks/index.md]: I like your use of the term "CDS Client" to refer to what we have previously been referring to as the "EHR". What do you think about opening a new issue to go through and replace EHR in our spec with "CDS Client"?

Comment (isaacvetter) [reply docs/hooks/index.md]: 👍 #407 created.

**Comment by isaacvetter** [comment]
@brettmarquard , @brynrhodes - this issue is important, would love your input.

File: docs/hooks/index.md [hunk 2]
```diff
@@ -147,6 +147,20 @@ Hook creators SHOULD include examples of the context.
...
+3 | Considered |  _The above, and …_ At least 3 distinct organizations recorded ten distinct implementer comments, including at least two CDS clients and three independent CDS services. The hook has been tested at two connectathons.
+4 | Documented | _The above, and …_ The author agrees that the artifact is sufficiently stable to require implementer consultation for subsequent non-backward compatible changes.  The hook is implemented in the standard CDS Hooks sandbox and multiple prototype projects. The Hook specification SHALL: <ul><ol>Identify a broad set of example contexts in which the hook may be used with a minimum of three, but as many as 8-10.</ol><ol>Clearly differentiate the hook from similar hooks or other standards to help an implementer determine if the hook is correct for their scenario.</ol><ol>Explicitly document example scenarios when the hook should not be used.</ol></ul>
```
Comment (brynrhodes) [inline docs/hooks/index.md]: Ten implementer comments seems like quite a high bar, I worry that that might not be reasonable to achieve? What if the hook definition just isn't controversial and there's no reason to comment?

File: docs/hooks/index.md [hunk 3]
```diff
@@ -147,6 +147,20 @@ Hook creators SHOULD include examples of the context.
...
+5 | Mature | _The above, and ..._ The hook has been implemented in production in at least two CDS clients and three independent CDS services. An HL7 working group ballots the hook and the hook has passed HL7 STU ballot.
```
Comment (brynrhodes) [inline docs/hooks/index.md]: Is HL7 balloting required for hooks? What would be the mechanism for that? An Implementation Guide? Or is it sufficient for the hook definition to achieve a given maturity rating in the hook registry? Do hooks get published to a separate place once they are mature?

File: docs/hooks/index.md (hunk 2 above)
Comment (brettmarquard) [reply docs/hooks/index.md]: Would requesting affirmative input be acceptable? I think two CDS clients and three independent CDS services giving it a thumbs up is a good idea.

File: docs/hooks/index.md (hunk 3 above)
Comment (brettmarquard) [reply docs/hooks/index.md]: Great question, I don't think a ballot is ever required. Hooks folks choose to ballot will get a special status in the hook registry

Comment (isaacvetter) [reply docs/hooks/index.md]: Added level 6 to better address.

**Pull request merged**